internal/context/ausf_context_init.go

```diff
@@ -23,13 +23,26 @@ func InitAusfContext(context *AUSFContext) {
 	context.GroupID = configuration.GroupId
 	context.NrfUri = configuration.NrfUri
 	context.NrfCertPem = configuration.NrfCertPem
-	context.UriScheme = models.UriScheme(configuration.Sbi.Scheme) // default uri scheme
-	context.RegisterIPv4 = factory.AusfSbiDefaultIPv4              // default localhost
-	context.SBIPort = factory.AusfSbiDefaultPort                   // default port
-	if sbi != nil {
-		if sbi.RegisterIPv4 != "" {
-			context.RegisterIPv4 = sbi.RegisterIPv4
+	context.SBIPort = factory.AusfSbiDefaultPort // default port
+	context.RegisterIPv6 = func () string {
+		if sbi.RegisterIPv6 != "" {
+			return sbi.RegisterIPv6
+		} else if sbi.RegisterIPv4 != "" {
+			return ""
+		} else {
+			return factory.AusfSbiDefaultIPv6 // default uri scheme
+		}
+	}()
+	context.RegisterIPv4 = func () string {
+		if sbi.RegisterIPv6 != "" {
+			return ""
+		} else if sbi.RegisterIPv4 != "" {
+			return sbi.RegisterIPv4
+		} else {
+			return factory.AusfSbiDefaultIPv6 // default uri scheme
```

Comment:
Default IPv6 on RegisterIPv4?

Reply:
If you have no sbi.RegisterIPv6 and no sbi.RegisterIPv4 set, you'll both have context.RegisterIPv4 and context.RegisterIPv6 be set?